62_CAD/shaders/main_pipeline/vertex_shader.hlsl

```diff
@@ -39,14 +40,18 @@ ClipProjectionData getClipProjectionData(in MainObject mainObj)
     }
 }
 
-float32_t2 transformPointScreenSpace(float64_t3x3 transformation, uint32_t2 resolution, float64_t2 point2d) 
+// TODO[Przemok]: Fix/Remove your todo here?!
+float2 transformPointScreenSpace(pfloat64_t3x3 transformation, uint32_t2 resolution, pfloat64_t2 point2d)
 {
-    float64_t2 ndc = transformPointNdc(transformation, point2d);
-    return (float32_t2)((ndc + 1.0) * 0.5 * resolution);
+    pfloat64_t2 ndc = transformPointNdc(transformation, point2d);
+    // TODO: "ndc + 1" doesn't work, probably because of `emulated_float64_t::create(float val)`
+    pfloat64_t2 result = (ndc + 1.0f) * 0.5f * _static_cast<pfloat64_t2>(resolution);
```

Comment:
+ int shouldn't work, I don't like it

and still because HLSL doesn't do conversion operators you're unable to get an implicit conversion to satisfy an argument type.

In C++ `double+int` compiles because `double` has an implicit constructor from `int` and only `operator+(const double)`

NOT because an overload of `+` exists for every other type (float, int, etc.) as that would wreck mayhem everywhere else (every function that expects a `double` would throw a fit if you try to feed an integer into it.

What I want to avoid is `double+int` working and then people asking why `f(double)` doesn't work when you call with `f(int)`